src/DotNet.Status.Web/TimelineIssueTriageService.cs

```diff
@@ -0,0 +1,286 @@
+        {
+            return _logic.GetTriageIssueProperty(propertyName, body);
+        }
+
+        private IList<TriageItem> GetTriageItemsProperties(string body)
+        {
+            return _logic.GetTriageItemsProperties(body);
+        }
+
+        public class TimelineIssueTriageLogic
+        {
+            // Search for triage item identifier information, example - "[BuildId=123456,RecordId=0dc87500-1d33-11ea-8b24-4baedbda8954,Index=0]"
+            private static readonly Regex _triageIssueIdentifierRegex = new Regex(@"\[BuildId=(?<buildid>[^,]+),RecordId=(?<recordid>[^,]+),Index=(?<index>[^\]]+)\]\s+\[Category=(?<category>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+            // Search for a triage item property, example = "[Category=foo]"
+            private static readonly Regex _triagePropertyRegex = new Regex(@"\[(?<key>[^=]+)=(?<value>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+
+            public IList<TriageItem> GetTriageItemsProperties(string body)
+            {
+                var items = new List<TriageItem>();
+
+                var matches = _triageIssueIdentifierRegex.Matches(body ?? "");
+
+                foreach (Match match in matches)
+                {
+                    TriageItem triageItem = new TriageItem();
+                    int.TryParse(match.Groups["buildid"].Value, out int buildId);
+                    int.TryParse(match.Groups["index"].Value, out int index);
+                    Guid.TryParse(match.Groups["recordid"].Value, out Guid recordId);
+                    var category = match.Groups["category"].Value;
+
+                    triageItem.BuildId = buildId;
+                    triageItem.RecordId = recordId;
+                    triageItem.Index = index;
+                    triageItem.UpdatedCategory = category;
+                    triageItem.ModifiedDateTime = DateTime.UtcNow;
+
+                    items.Add(triageItem);
+                }
+
+                return items;
+            }
+
+            public string GetTriageIssueProperty(string propertyName, string body)
+            {
+                MatchCollection propertyMatches = _triagePropertyRegex.Matches(body);
+                foreach (Match propertyMatch in propertyMatches)
+                {
+                    if (propertyMatch.Success)
+                    {
+                        string key = propertyMatch.Groups["key"].Value;
+                        string value = propertyMatch.Groups["value"].Value;
+                        if (propertyName.Equals(key, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return value;
+                        }
+                    }
+                }
+                return null;
+            }
+
+            public bool IsDuplicate(IList<TriageItem> triageItems, IList<TriageItem> existingIssueItems)
```

Comment:
Not sure this method belongs to this class? I think it's tied to the logic in the other?